### ISSUE TITLE
Conflicting dependences where different Python versions are specified.   Because usignals depends on both ipython (^5) and ipython (^7), version solving failed.

### ISSUE THREAD

**Issue by kierun** [sdispater/poetry#1480]
- [X] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [X] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [X] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: CentOS 7, Python 3.7.4
- **Poetry version**: 0.12.17
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: 

## Issue

I have this so that my module works on either 2.7 or 3.7 and in `pyproject.toml`, I specified:

```
[tool.poetry.dependencies]
python = "^2.7 || ^3.7"  # ← Run on either 2.7 or 3.7???
[…]
[tool.poetry.dev-dependencies]
ipython = [
    {version = "^5", python = "~2.7"},  # ← Install version 5 under py2.7???
    {version = "^7", python = ">=3.5"}  # ← Install version 7.8 under py37???
]
```

Which I assume means that I want ipython version 5 for 2.7 and ipython version 7 for any python above 3.5. However, what I get is:

```
poetry -vvv debug:resolve
Resolving dependencies...
[…]
   1: fact: usignals depends on ipython (^5)
   1: fact: usignals depends on ipython (^7)
[…]
   1: derived: ipython (^7.8)
   1: conflict: usignals depends on ipython (^5)
   1: ! not ipython (^5) is satisfied by ipython (^7)
   1: ! which is caused by "usignals depends on ipython (^7)"
   1: ! thus: version solving failed
   1: Version solving took 0.016 seconds.
   1: Tried 1 solutions.
```

_Is this a bug or did I misunderstand something?_

**Comment by kierun** [comment]
WOOT! Thank you.

Is there a way I can test this on my setup? If not, when is the next release out

**Comment by agilgur5** [comment]
@kierun looks like this was released in [`1.0.0b3`](https://github.com/sdispater/poetry/releases/tag/1.0.0b3)
